index.js

```diff
@@ -39,355 +32,291 @@ module.exports = Store;
+  json(replacer = null, space = this.indent) {
+    return JSON.stringify(this.data, replacer, space);
   }
 
-  options = options || {};
-
-  if (keys) {
-    keys.forEach(function(key) {
-      proto.del.call(this, key);
-    }.bind(this));
-    this.save();
+  /**
+   * Persist the store to the file system.
+   *
+   * ```js
+   * store.save();
+   * ```
+   * @api public
+   */
+
+  save() {
+    mkdirSync(path.dirname(this.path), this.options.mkdir);
+    fs.writeFileSync(this.path, this.json(), { mode: 0o0600 });
     return this;
   }
 
-  if (options.force !== true) {
-    throw new Error('options.force is required to delete the entire cache.');
+  /**
+   * Load the store.
+   * @return {Object}
+   */
+
+  load() {
+    try {
+      return JSON.parse(fs.readFileSync(this.path));
+    } catch (err) {
+      if (err.code === 'EACCES') {
+        err.message += '\ndata-store does not have permission to load this file\n';
+        throw err;
+      }
+      if (err.code === 'ENOENT' || err.name === 'SyntaxError') {
+        this.data = {};
+        return {};
+      }
+    }
   }
 
-  keys = Object.keys(this.cache);
-  this.clear();
-
-  // if no keys are passed, delete the entire store
-  utils.del.sync(this.path, options);
-  keys.forEach(function(key) {
-    this.emit('del', key);
-  }.bind(this));
-  return this;
-};
+  get keys() {
+    return Object.keys(this.data);
+  }
+}
 
 /**
- * Returns an array of all Store properties.
+ * Create a directory and any intermediate directories that might exist.
  */
 
-utils.define(Store.prototype, 'keys', {
-  configurable: true,
-  enumerable: true,
-  set: function(keys) {
-    utils.define(this, 'keys', keys);
-  },
-  get: function fn() {
-    if (fn.keys) return fn.keys;
-    fn.keys = [];
-    for (var key in this) fn.keys.push(key);
-    return fn.keys;
+function mkdirSync(dirname, options = {}) {
+  assert.equal(typeof dirname, 'string', 'expected a string');
+  const opts = Object.assign({ cwd: process.cwd() }, options);
+  const segs = path.relative(opts.cwd, dirname).split(path.sep);
+  for (let i = 0; i <= segs.length; i++) {
+    try {
+      fs.mkdirSync(path.join(opts.cwd, ...segs.slice(0, i)), mode(opts));
+    } catch (err) {
+      if (err.code !== 'EEXIST') {
+        throw err;
+      }
+    }
   }
-});
-
-/**
- * Define a non-enumerable property on the instance.
- *
- * @param {String} `key`
- * @param {any} `value`
- * @return {Object} Returns the instance for chaining.
- * @api public
- */
+  return dirname;
+}
 
-Store.prototype.define = function(key, value) {
-  utils.define(this, key, value);
-  return this;
-};
+function get(obj = {}, prop = '') {
+  return obj[prop] || split(prop).reduce((acc, k) => acc && acc[strip(k)], obj);
+}
 
-/**
- * Read JSON files.
- *
- * @param {String} `fp`
- * @return {Object}
- */
+function set(obj = {}, prop = '', val) {
+  return split(prop).reduce((acc, k, i, arr) => {
+    return (acc[k] = arr.length - 1 > i ? (acc[k] || {}) : val);
```

Comment:
When `acc[k]` exists, I think if it's not an object, then an object should be set on `acc[k]`, then returned.